secrets/secrets_test.go

```diff
@@ -31,6 +31,24 @@ func TestLoadAllRules_DuplicateRuleID(t *testing.T) {
 	}
 }
 
+func TestLoadAllRules_CustomRulesLoaded(t *testing.T) {
+	allRules, err := loadAllRules()
+	ruleIDMap := make(map[string]bool)
+	if err != nil {
+		t.Error(err)
+	}
+
+	for _, rule := range allRules {
+		ruleIDMap[rule.Rule.RuleID] = true
+	}
+	for _, customRule := range customRules {
+		_, ok := ruleIDMap[customRule.RuleID]
+		if !ok {
+			t.Errorf("custom rule not found: %s", customRule.RuleID)
+		}
+	}
+}
+
```

Comment:
I feel this test is tightly coupled with the implementation, you're taking both lists and tests they are joined, so you test almost nothing.

I suggest a higher level of test:
https://github.com/Checkmarx/2ms/blob/missing-secrets/secrets/secrets_test.go#L361
You can take my test from the other branch (take only the passing examples), and test your custom rule. Because this is your task- make this rule to work, and we don't care how you achieved that.